src/main/scala/com/fulcrumgenomics/bam/ClipOverlappingReads.scala

```diff
@@ -67,8 +67,7 @@ class ClipOverlappingReads
   Io.assertReadable(ref)
   Io.assertCanWriteFile(output)
 
-  private val clippingMode = if (softClip) ClippingMode.Soft else ClippingMode.Hard
-  private val clipper = new SamRecordClipper { override val autoClipAttributes = ClipOverlappingReads.this.autoClipAttributes }
+  private val clipper = new SamRecordClipper(mode=if (softClip) ClippingMode.Soft else ClippingMode.Hard, autoClipAttributes=autoClipAttributes)
```

Comment:
I'd prefer not to do that because a) there are three clipping modes, and b) in the two tools where I use this I have different defaults, and hence different booleans (one is `softClip` the other is `hardClip`).